### PR TITLE
Preparation release v2022.1 (Backend)

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

For release.

## Changes

Changes "Changes", "LDNS.pm" and "Makefile.PL".

## Dependency

Commit 66b5fc979c6e5e20f1e60d50867e2cbaff30e4ed assumes that #1037 is also merged. Else it should be removed before merge.

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -1,5 +1,70 @@
...
+
+ [Breaking changes]
+ - Domain names cannot be entered with consecutive tailing
+   dots (#983)
+
+ [Features]
+ - Adds support for CentOS Linux 7 again - to be removed again
+   with release v2023.1 (#1027)
+ - Updates zmb for "start_domain_test" and "add_batch_job" to include
+   support for "queue"(#1006, #1002)
```
Comment (tgreenx) [inline Changes:32]: Missing space between `"queue"` and following parenthesis

File: Changes [hunk 2]
```diff
@@ -1,5 +1,70 @@
...
+ - Improvs logging (#1014, #1013, #1004, #996, #966)
+ - Improves test progress computation(#988)
+ - Improves reuse of previous test (#1010, #979)
+ - Makes sure domain names are stored without trailing dot even if
+   submitted with trailing dot (#1005)
+ - Improved documentation (#1007, #993, #737, #974, #976, #968, #950)
+ - Fixes handling of trailing dots in domain names and name server
+   names. Consecutive dots are not permitted. Name server name with
+   trailing dot will not crash. (#983)
+ - Fixes bug that prevented domain "0" to be tested (#988)
```
Comment (tgreenx) [inline Changes:52]: Typo: `#988` ->`#998`

File: Changes [hunk 3]
```diff
@@ -1,5 +1,70 @@
...
+ - Make batch tests do not update progress during test (#944, #274)
+ - Improved database schema (#977, #970)
+ - Uses ISO 8601 format on all datetimes (#967)
+ - Makes call for non-existing batch ID return error (#965, #860)
+ - Improves IDN support (#963)
+ - Makes Testagent survive database connection failure (#955, #878)
+ - Improves error handling in TestAgent(#956, #933)
+ - Cleans up determination of config file default path (#937)
+ - Fixes unstoppable Testagent daemon (#943, #905)
+
```
Comment (tgreenx) [inline Changes:67]: I believe #769 #971 are missing from this list (- they aren't related to Travis).
Also, #947 #961 #982 #1003 aren't either but they relate to unit tests fixes so I suppose it is intentional?

File: Changes [hunk 4]
```diff
@@ -1,5 +1,70 @@
...
+ [Features]
+ - Adds support for CentOS Linux 7 again - to be removed again
+   with release v2023.1 (#1027)
+ - Updates zmb for "start_domain_test" and "add_batch_job" to include
+   support for "queue"(#1006, #1002)
+ - Makes "/" valid character in domain names and fixes handling of
+   "_" in domain names (#969)
+
+ [Fixes]
+ - Adds migration script and instruction for database for
```
Comment (tgreenx) [inline Changes:37]: `instruction ` -> `instructions`

File: Changes [hunk 5]
```diff
@@ -1,5 +1,70 @@
...
+   this version (#031, #1030,
+ - Fixes libraries and dependencies (#1029, #1022, #991, #990)
+ - Updates language translations (#1028, #1019, #1024, #1020,
+   #1025, #1018, #1016)
+ - Updates installation instructions (#1021, #1023)
+ - Improvs logging (#1014, #1013, #1004, #996, #966)
```
Comment (tgreenx) [inline Changes:43]: `Improvs ` -> `Improves`

File: Changes [hunk 6]
```diff
@@ -1,5 +1,70 @@
...
+ - Improves test progress computation(#988)
```
Comment (tgreenx) [inline Changes:44]: Missing space between `computation` and following parenthesis

File: Changes [hunk 7]
```diff
@@ -1,5 +1,70 @@
...
+ - Adds support for CentOS Linux 7 again - to be removed again
+   with release v2023.1 (#1027)
+ - Updates zmb for "start_domain_test" and "add_batch_job" to include
+   support for "queue"(#1006, #1002)
+ - Makes "/" valid character in domain names and fixes handling of
+   "_" in domain names (#969)
+
+ [Fixes]
+ - Adds migration script and instruction for database for
+   this version (#031, #1030,
```
Comment (blacksponge) [inline Changes:38]: > this version (#031, #1030,

If feel that the PR / issue list has a problem here.

File: Changes [hunk 8]
```diff
@@ -1,5 +1,70 @@
...
+ - Updates messages (#986)
+ - Uses libidn2 instead of libidn (#984)
+ - Updates zmb (#985, #920)
+ - Make batch tests do not update progress during test (#944, #274)
+ - Improved database schema (#977, #970)
+ - Uses ISO 8601 format on all datetimes (#967)
+ - Makes call for non-existing batch ID return error (#965, #860)
+ - Improves IDN support (#963)
+ - Makes Testagent survive database connection failure (#955, #878)
+ - Improves error handling in TestAgent(#956, #933)
```
Comment (tgreenx) [inline Changes:64]: Missing space between `TestAgent` and following parenthesis

File: Changes (hunk 5 above)
Comment (blacksponge) [inline Changes:43]: Improvs -> Improves

File: Changes (hunk 6 above)
Comment (blacksponge) [inline Changes:44]: Missing space before `(#988)`

File: Changes [hunk 9]
```diff
@@ -1,5 +1,70 @@
...
+   be removed with release v2023.1. Use the "created_at"
+   key instead. See "docs/API.md#api-method-get_test_results"
+   and "docs/API.md#api-method-get_test_history".
+ - API method "get_test_results" returns key "id", which is now
+   deprecated and will be removed in v2022.2 release. See
+   "docs/API.md#api-method-get_test_results". (#949)
+ - Removes deprecated configuration options (#954)
+
+ [Breaking changes]
+ - Domain names cannot be entered with consecutive tailing
```
Comment (tgreenx) [inline Changes:25]: `tailing` -> `trailing`

**Review by tgreenx — CHANGES_REQUESTED** [review]
Some comments. Also, one remark: Zonemaster-CLI major version has been bumped due to the same "Breaking change" present in this repo, but Zonemaster-Backend hasn't. To keep consistency we should either: 1) revert Zonemaster-CLI major version bump if we consider this isn't enough of a "Breaking change", or 2) bump Zonemaster-Backend to 9.0.0

File: Changes (hunk 7 above)
Comment (matsduf) [reply Changes:38]: Good catch. It is a missing "1" (#1031).

File: Changes (hunk 5 above)
Comment (matsduf) [reply Changes:43]: Fixed.

File: Changes (hunk 6 above)
Comment (matsduf) [reply Changes:44]: Fixed.

File: Changes (hunk 2 above)
Comment (matsduf) [reply Changes:52]: Good catch! Fixed.

File: Changes (hunk 1 above)
Comment (matsduf) [reply Changes:32]: Fixed.

**Comment by matsduf** [comment]
> Some comments. Also, one remark: Zonemaster-CLI major version has been bumped due to the same "Breaking change" present in this repo, but Zonemaster-Backend hasn't. To keep consistency we should either: 1) revert Zonemaster-CLI major version bump if we consider this isn't enough of a "Breaking change", or 2) bump Zonemaster-Backend to 9.0.0

It is an API change, so I found it correct to bump Zonemaster-CLI to that version. I would have bumped Zonemaster-Backend too if it weren't for update scripts. Maybe it is not too complicated. I will take a look at it. After fixing these little things.

File: Changes (hunk 5 above)
Comment (matsduf) [reply Changes:43]: Fixed.

File: Changes (hunk 9 above)
Comment (matsduf) [reply Changes:25]: Fixed.

**Review by tgreenx — APPROVED** [review]

File: Changes [hunk 10]
```diff
@@ -1,5 +1,70 @@
...
+ - Fixes handling of trailing dots in domain names and name server
+   names. Consecutive dots are not permitted. Name server name with
+   trailing dot will not crash. (#983)
+ - Fixes bug that prevented domain "0" to be tested (#998)
+ - Improves code (#997, #981, #980, #957, #919, #975, #973, #972,
+   #964, #962, #958, #942, #888, #769)
+ - Updates messages (#986)
+ - Uses libidn2 instead of libidn (#984)
+ - Updates zmb (#985, #920)
+ - Make batch tests do not update progress during test (#944, #274)
```
Comment (mvw-afnic) [inline Changes:58]: I’d suggest: “Batch tests no longer update progress while running”. Or “Make batch test run without updating progress” maybe?

**Review by mvw-afnic — CHANGES_REQUESTED** [review]

File: Changes (hunk 10 above)
Comment (matsduf) [reply Changes]: Fixed.

**Review by mvw-afnic — APPROVED** [review]
Looks good to me.

**Pull request merged**